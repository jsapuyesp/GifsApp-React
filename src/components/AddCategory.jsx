import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = (props) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim() < 1) return;

    // props.setCategories((categories) => [inputValue, ...categories]);
    props.onNewValue(inputValue.trim());

    setInputValue("");
  };

  return (
    <form onSubmit={(event) => onSubmit(event)}>
      <input
        type="text"
        placeholder={props.placeholder}
        value={inputValue}
        onChange={(event) => onInputChange(event)}
      />
    </form>
  );
};

AddCategory.propTypes = {
  placeholder: PropTypes.string.isRequired,
};
