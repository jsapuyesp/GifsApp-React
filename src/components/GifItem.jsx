import PropTypes from "prop-types";

export const GifItem = (props) => {
  let url = props.url;
  let title = props.title;
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
