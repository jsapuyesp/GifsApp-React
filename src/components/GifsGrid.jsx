import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import PropTypes from "prop-types";

export const GifsGrid = (props) => {
  const { images, isLoading } = useFetchGifs(props.category);
  const title =
    props.category.charAt(0).toUpperCase() + props.category.slice(1);

  return (
    <>
      <h3>{title}</h3>
      {isLoading && <h2>Cargando...</h2>}

      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifsGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
