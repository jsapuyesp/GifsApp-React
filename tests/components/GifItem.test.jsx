import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem/>', () => {
  const image = {
    id: 0,
    title: 'Titulo',
    url: 'http://localhost/url'
  }
  test('Debe hacer match con el snapshot', () => {

    const { container } = render(<GifItem key={image.id} {...image} />)
    expect(container).toMatchSnapshot();
  });

  test('El título y url de la imagen deben coincidir', () => {

    render(<GifItem key={image.id} {...image} />)
    // screen.debug();
    const { src, alt } = screen.getByRole('img');
    expect(alt).toBe(image.title);
    expect(src).toBe(image.url);
  });

  test('Se debe renderizar el título', () => {

    render(<GifItem key={image.id} {...image} />)
    // screen.debug();

    expect(screen.getByText(image.title)).toBeTruthy();
  });



});