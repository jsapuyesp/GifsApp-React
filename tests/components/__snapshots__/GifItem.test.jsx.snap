// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo"
      src="http://localhost/url"
    />
    <p>
      Titulo
    </p>
  </div>
</div>
`;
